wallet/README.md

```diff
@@ -0,0 +1,56 @@
+
+Note that all the port numbers and bindings for the RPC, whether for the node or the wallet, are fully configurable. Run the node or wallet executables with `--help` option to see how to do that. The `bind` ports is what you are looking for in that case.
+
+
+### Security
+
+The security of the wallet and node is the responsibility of the node operator. Mintlayer does not bear any of the responsibility for the mismanagement of a node. For example, binding to the network address `0.0.0.0` without a firewall is an open invitation for bots to brute-force the server and attempt to break it. Please exercise good security practices when managing a node. More information can be found [here](/build-tools/linux-systemd-service/README.md).
+
+### Modes of operation for the wallet
+
+At a time, a wallet file can only be open by one program.
+
+There are two distinct executables for the wallet:
+
+#### wallet-cli
+
+The command-line interface of the wallet. The wallet-cli can be simply started by running it, and a wallet can be created with the `wallet-create` or an existing wallet file opened with the command `wallet-open`. A wallet can also be opened automatically on launch using command line arguments. Run the wallet-cli with the `--help` command line argument to see what options are available.
+
+The wallet-cli can be run in many modes. Please consult the `--help` command-line arguments to see how each of these are achieved. We do not list explicit command-line arguments here to ensure that information is never outdated. Here are the modes of operation:
+
+1. It can be run as a simple wallet-file manager. It creates/opens a file, and the user interacts with it by keyboard.
+2. In addition to a simple wallet-file manager, it can also be used as an RPC server (enabled with a command line argument), allowing the user to access the wallet from the outside.
+3. Instead of the previous options, it can be used as a lightweight controller of a remote RPC wallet. Hence, you can run an RPC wallet, and get the wallet-cli to connect to that wallet using RPC, and use the easy commands of the wallet-cli to control that wallet. Notice that this option excludes the other ones.
+
+#### wallet-rpc-daemon
+
+The server that can be used as a service. The wallet-cli cannot be used as a systemd service, due to its requirement to have keyboard interactions. Hence, `wallet-rpc-daemon` can be set up on servers as a service.
+
+The RPC for the wallet (whether from the daemon or cli) has the default port values 3034 for mainnet and 13034 for testnet. There are different modes of authentication similar to those of the node. Run the `wallet-rpc-daemon` with the command-line argument `--help` to see the available options.
+
+The `wallet-rpc-daemon` requires a file to run. Once a file is open, the wallet can be controlled remotely with RPC. Any http and websocket RPC client can communicate with the RPC daemon. To see examples, [visit this readme file](wallet-rpc-daemon/README.md).
```

Comment:
this requirement was removed, it can work just like the cli wallet it can start without a wallet and later create or open a wallet.

Reply:
If this requirement was removed, we should make the file use a `wallet-file` command-line argument... otherwise it's confusing. I'm actually not sure it's possible in the current form.

Reply:
yes we should make the argument named same as the cli.
but yes it works, the functional tests do that start the rpc daemon and then create a wallet, as the commands in the test are the same for cli and rpc wallet.